pyart/util/columnsect.py

```diff
@@ -0,0 +1,294 @@
+        the various fields within the radar object, as well as,
+        the height gates above that point.
+
+    Dictionary keys:
+    ----------------
+    date : string
+        date of the radar scan in YYYY-MM-DD format
+    time : string
+        beginning time of each scan in HH:MM:SSZ format
+    xgate : list
+        East-west distance to target in m
+    ygate : list
+        North-South distance to target in m
+    zgate : list
+        Height above target in m
+    tardis : float
+        Distance from radar to target in m
+    tarazi : float
+        Forward azimuth angle from radar to target in degrees
+    moments : list
+        Radar fields for the column above the target.
+        Dependent on which radar fields are within the
+        input radar object.
+
+    """
+    # Make sure lat, lons are valid
+    check_lat(lat)
+    check_lon(lon)
+    # initiate a dictionary to hold the striped out meat data from each scan
+    meta = {'date': [], 'time': [], 'xgate': [], 'ygate': [],
+            'zgate': [], 'distance': [], 'azimuth': []
+            }
+    # initiate a diciontary to hold the moment data.
+    moment = {key: [] for key in radar.fields.keys()}
+    # call the sphere_distance function
+    dis = sphere_distance(radar.latitude['data'][0],
+                          lat,
+                          radar.longitude['data'][0],
+                          lon)
+    # call the for_azimuth function
+    azim = for_azimuth(radar.latitude['data'][0],
+                       lat,
+                       radar.longitude['data'][0],
+                       lon)
+    # call the get_column_ray function
+    ray = get_column_rays(radar, azim)
+    # add the parameters to the meta dictionary
+    meta['date'].append(radar.time['units'].split(' ')[2].split('T')[0])
+    meta['time'].append(radar.time['units'].split(' ')[2].split('T')[-1])
+    meta['distance'].append(dis / 1000.)
+    meta['azimuth'].append(azim)
+    # Determine the gates for the rays
+    (rhi_x,
+     rhi_y,
+     rhi_z) = antenna_vectors_to_cartesian(radar.range['data'],
+                                           radar.azimuth['data'][ray],
+                                           radar.elevation['data'][ray],
+                                           edges=True)
+    # Calculate distance from the x,y coordinates to target
+    rhidis = np.sqrt((rhi_x**2) + (rhi_y**2)) * np.sign(rhi_z)
+    for i in range(len(ray)):
+        tar_gate = np.argmin(abs(rhidis[i, 1:] - (dis)))
+        for key in moment:
+            moment[key].append(radar.fields[key]['data'][ray[i], tar_gate])
+        # fill the meta dictionary for further analysis
+        meta['xgate'].append(rhi_x[i, tar_gate])
+        meta['ygate'].append(rhi_y[i, tar_gate])
+        meta['zgate'].append(rhi_z[i, tar_gate])
+    # Combine the meta and moment data
+    column = dict(meta)
+    column.update(moment)
+
+    return column
```

Comment:
this would be a good point to convert to either an Xarray.Dataset or a pyart.core.Radar object...